spec/models/ext_management_system_spec.rb

```diff
@@ -396,6 +396,96 @@
     end
   end
 
+  context "#pause!" do
+    before { Zone.seed }
+
+    it 'disables an ems with child managers and moves parent to maintenance zone' do
+      zone  = FactoryGirl.create(:zone)
+      ems   = FactoryGirl.create(:ext_management_system, :zone => zone)
+      child = FactoryGirl.create(:ext_management_system, :zone => zone)
+      ems.child_managers << child
+
+      ems.pause!
+
+      expect(ems.enabled).to be_falsy
+      expect(ems.zone).to eq(Zone.maintenance_zone)
+      expect(ems.zone_before_pause).to eq(zone)
+
+      child.reload
+      expect(child.enabled).to be_falsy
+    end
+  end
+
+  context "#resume" do
+    before { Zone.seed }
+
+    it "enables an ems with child managers and move parent from maintenance zone" do
+      zone = FactoryGirl.create(:zone)
+      ems = FactoryGirl.create(:ext_management_system,
+                               :zone_before_pause => zone,
+                               :zone              => Zone.maintenance_zone,
+                               :enabled           => false)
+
+      child = FactoryGirl.create(:ext_management_system,
+                                 :zone    => zone,
+                                 :enabled => false)
+      ems.child_managers << child
+
+      ems.resume!
+
+      expect(ems.enabled).to eq(true)
+      expect(ems.zone).to eq(zone)
+      expect(ems.zone_before_pause).to be_nil
+
+      child.reload
+      expect(child.enabled).to eq(true)
+    end
+
+    it "doesn't change zone when ems was disabled before" do
+      zone = FactoryGirl.create(:zone)
+      ems = FactoryGirl.create(:ext_management_system,
+                               :zone_before_pause => nil,
+                               :zone              => zone,
+                               :enabled           => false)
+
+      ems.resume!
+
+      expect(ems.enabled).to eq(true)
+      expect(ems.zone).to eq(zone)
+    end
+  end
+
+  context "changing zone" do
+    before { Zone.seed }
+
+    it 'is allowed when enabled' do
+      zones = (1..2).collect { FactoryGirl.create(:zone) }
```

Comment:
This completely works but `2.times.map {}` is a little nicer to read

Reply:
or `FactoryGirl.create_list(:zone, 2)`

Reply:
:+1: nice didn't know that one

Reply:
sorry but aren't obvious all of these approaches? 
Is this supposed to be longest PR ever? :)

Reply:
and for me, it's more readable `2.times`, or `collect` than `create_list`, what is create_list? Array, Hash, Set? I have to use google to understand, waste of time :/